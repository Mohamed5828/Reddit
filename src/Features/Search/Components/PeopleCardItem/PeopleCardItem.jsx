import { Container } from "./PeopleCardItem.styled";
import { useState } from "react";
// import logo from "../../Assets/download.jpg";
// Import api
import axios from "API/axios";
// import useFetch from "Hooks/useFetch";
import { Link } from "react-router-dom";
import useFetchFunction from "Hooks/useFetchFunction";
/**
 * Component that contains the PeopleCardItem and manage the state of the button Follow.
 *
 * @Component
 * @param {String} avatar - The image of the PeopleCardItem
 * @param {String} userID - The ID of the PeopleCardItem
 * @param {String} about - The Description of the PeopleCardItem
 * @param {number} totalKarmas - The totalKarmas of the PeopleCardItem
 * @param {String} username - The name of the PeopleCardItem
 * @returns {React.Component}
 */
const PeopleCardItem = ({
  avatar,
  userID,
  about,
  totalKarmas,
  username,
  isFollow,
}) => {
  const [joinRes, errorJoin, joinLoading, fetchFunction] = useFetchFunction();
  const [isJoinedstate, setisJoined] = useState(false);
  const joinCommunity = (btnState) => {
    fetchFunction({
      axiosInstance: axios,
      method: "POST",
      url: "http://localhost:8000/follow",
      requestConfig: {
        headers: {
          "Content-Language": "en-US",
        },
        data: {
          action: btnState ? "Follow" : "unFollow",
          userName: `${username.slice(1)}`,
        },
      },
    });
  };
  // People Subscriptions

  // find if the user following this one

  // initialState for the following operations
  const initialState = `${isFollow !== undefined ? "Following" : "Follow"}`;
  // the state of the buuton
  const [btnContent, setBtnContent] = useState(
    isFollow !== undefined ? "Following" : "Follow"
  );

  /**
   * it is the function that handle the state of the button when click on it.
   *
   * @param {Object} e it is an object which i can preventDefault and not redirect to the link
   */
  function clickHandler(e) {
    e.preventDefault();
    if (btnContent === "Follow") {
      setBtnContent("Following");
    } else {
      setBtnContent("Follow");
    }
    ///////////////////
    let btnState;
    if (btnContent === "Follow") {
      btnState = true;
    } else {
      btnState = false;
    }
    joinCommunity(btnState);
    ////////////////
  }

  /**
   * it is the function that handle the state of the button when mouseEnter on it.
   *
   */
  function mouseEnterHandler() {
    if (btnContent === "Following") {
      setBtnContent("Unfollow");
    }
  }

  /**
   * it is the function that handle the state of the button when MouseLeave on it.
   *
   */
  function MouseLeaveHandler() {
    if (btnContent === "Unfollow") {
      setBtnContent("Following");
    }
  }
  return (
    <Container>
      <Link to="#">
        <div className="item">
          <img src={require(`../../Assets/${avatar}`)} alt="userImage" />
          <div className="info">
            <div className="info2">
              <h6>{username.slice(1)}</h6>
              <p>{totalKarmas}m Karma</p>
            </div>
          </div>
          <div className="button">
            <button
              onClick={clickHandler}
              onMouseEnter={mouseEnterHandler}
              onMouseLeave={MouseLeaveHandler}
              className="follow"
            >
              {btnContent}
            </button>
          </div>
        </div>
      </Link>
    </Container>
  );
};

export default PeopleCardItem;
