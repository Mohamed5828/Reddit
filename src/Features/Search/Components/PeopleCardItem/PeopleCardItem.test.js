import { shallow } from "enzyme";
import PeopleCardItem from "./PeopleCardItem";
describe("People Card Item", () => {
  it("this is a test for People Card Item", () => {
    expect(
      shallow(
        <PeopleCardItem
          avatar={"People_Image.jpg"}
          userID={"t5_imagepro"}
          about={
            "A subreddit dedicated to German photos and portraits from the period of 1933-1946 (dates are flexible)"
          }
          totalKarmas={10}
          username={"/u/Abdo"}
        />
      )
    ).toMatchSnapshot();
  });
});
// { avatar, userID, about, totalKarmas, username }
