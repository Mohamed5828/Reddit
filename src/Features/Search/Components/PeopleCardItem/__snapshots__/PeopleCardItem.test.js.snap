// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`People Card Item this is a test for People Card Item 1`] = `
<styled.div>
  <Link
    to="#"
  >
    <div
      className="item"
    >
      <img
        alt="userImage"
        src="People_Image.jpg"
      />
      <div
        className="info"
      >
        <div
          className="info2"
        >
          <h6>
            u/Abdo
          </h6>
          <p>
            10
            m Members
          </p>
        </div>
      </div>
      <div
        className="button"
      >
        <button
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          Following
        </button>
      </div>
    </div>
  </Link>
</styled.div>
`;
